### PR TITLE
posix: abstract pthread_cond_t as uint32_t

### PR DESCRIPTION
**Pull request by cfriedt**
4 commits

* lib: posix: internal: use a more generic INIT mask and inlines
* posix: pthread: take care with pthread cond resources
* posix: cond: abstract pthread_cond_t as uint32_t
* tests: posix: cond: test to ensure there is no resource leakage

See individual commit messages for details.

### CODE REVIEW

**Comment by cfriedt** [comment]
This will be rebased after #52087 is merged, just keeping in draft for now.

**Comment by cfriedt** [comment]
Compliance check failures are a false positive (pthread functions return non-negative errno values)

**Review by jeremybettis — DISMISSED** [review]

**Review by jeremybettis — DISMISSED** [review]

**Comment by cfriedt** [comment]
@stephanosio - are you able to take a look?

**Review by stephanosio — DISMISSED** [review]

**Comment by cfriedt** [comment]
@stephanosio, @jeremybettis - should be ready for round 2, if you have a moment.

File: lib/posix/pthread_cond.c [hunk 1]
```diff
@@ -8,18 +8,88 @@
...
+ */
+BUILD_ASSERT(CONFIG_MAX_PTHREAD_COND_COUNT < PTHREAD_COND_MASK_INIT,
+	     "CONFIG_MAX_PTHREAD_COND_COUNT is too high");
+
+static inline size_t posix_cond_to_offset(struct posix_cond *cv)
+{
+	return cv - posix_cond_pool;
+}
+
+static inline size_t to_posix_cond_idx(pthread_mutex_t cond)
```
[the remark this answers is not in the document]
Comment (cfriedt) [reply lib/posix/pthread_cond.c]: Fixed!

File: lib/posix/posix_internal.h [hunk 2]
```diff
@@ -12,6 +12,7 @@
  * verified (against internal status) in lock / unlock / destroy functions.
  */
 #define PTHREAD_MUTEX_MASK_INIT 0x80000000
+#define PTHREAD_COND_MASK_INIT	PTHREAD_MUTEX_MASK_INIT
```
[the remark this answers is not in the document]
Comment (cfriedt) [reply lib/posix/posix_internal.h]: Fixed!

**Review by stephanosio — APPROVED** [review]

**Comment by cfriedt** [comment]
@enjiamai @KangJianX - any reviews from you?

**Comment by cfriedt** [comment]
@jeremybettis - IIRC you had previously approved this one. Will you re-ack?

**Comment by cfriedt** [comment]
@keith-packard would you maybe ack? This is blocking a few PRs still..

**Review by mbolivar-nordic — APPROVED** [review]

**Review by jeremybettis — APPROVED** [review]

**Pull request merged**